lib/aasm/configuration.rb

```diff
@@ -28,5 +28,11 @@ class Configuration
 
     # namespace reader methods and constants
     attr_accessor :namespace
+
+    # These are the methods that will be override in a sub-class.
+    attr_accessor :override_methods
+
+    # Ignore all warnings, to be used at your own risk.
+    attr_accessor :ignore_override_warnings
```

Comment:
I'm really afraid that this this will become a mainstream aasm definition and people will still be overriding methods silently.
I would go even further and would raise an exception (a cumulative one after all definitions are compiled) in development environment to force to either ignore the warnings or to fix the names of the state/transition/concurrent property/method.